doc/dev/testing.md

```diff
@@ -2,17 +2,37 @@
 
 You can run the unit tests after completing a build using the following:
 
+After [building lean](../make/index.md) you can run all the tests using
+```
+cd build/release
+make test ARGS=-j4
+```
+
+Change the 4 to the maximum number of parallel tests you want to
+allow. The best choice is the number of CPU cores on your machine as
+the tests are mostly CPU bound.  You can find the number of processors
+on linux using `cat /proc/cpuinfo | grep processor | wc -l` and on
+Windows it is the NUMBER_OF_PROCESSORS environment variable.
+
+You can run tests after [building a specific stage](bootstrap.md) by
+adding the `-C stageN` argument. Most of the tests are in stage1.  The
+Lean tests will automatically use that stage's corresponding Lean
+executables
+
+You can also use `ctest` directly if you are in the right folder.  So
+to run stage1 tests with a 300 second timeout run this:
+
 ```bash
-cd build/release/stage0
+cd build/release/stage1
 ctest -j 4 --output-on-failure --timeout 300
 ```
 
-You can increase the parallelism from 4 to as many cores you have
-available to decrease overall test time.  You can also increase the timeout
-from 300 seconds if you need to.
+To get verbose output from ctest pass the `-VV` command line option.

```

Comment:
Interesting, I don't think I've ever used that. When should we use it?

Reply:
In a quick test, the output from `-V` and `-VV` looks identical to me

Reply:
The [documentation ](https://cmake.org/cmake/help/v3.0/manual/ctest.1.html) says 

- Enable more verbose output from tests.
   Test output is normally suppressed and only summary information is displayed. This option will show _even more_ test output.

But it could be that our style of testing there is no "even more" output available... so I updated the doc to just mention `--verbose` and what that does exactly.

Reply:
for the record I did find some differences when comparing the entire output with -V versus -VV :-) 

![image](https://user-images.githubusercontent.com/18707114/134433688-b64633d0-b571-4cf2-b857-d25cc19e021b.png)

Reply:
I think that's just `make` skipping work on a repeated run